AsyncDisplayKit/Details/ASPINRemoteImageDownloader.m

```diff
@@ -114,7 +200,19 @@ - (void)setPriority:(ASImageDownloaderPriority)priority withDownloadIdentifier:(
       pi_priority = PINRemoteImageManagerPriorityVeryHigh;
       break;
   }
-  [[PINRemoteImageManager sharedImageManager] setPriority:pi_priority ofTaskWithUUID:downloadIdentifier];
+  [[self sharedPINRemoteImageManager] setPriority:pi_priority ofTaskWithUUID:downloadIdentifier];
+}
+
+#pragma mark - PINRemoteImageManagerAlternateRepresentationProvider
+
+- (id)alternateRepresentationWithData:(NSData *)data options:(PINRemoteImageManagerDownloadOptions)options
```

Comment:
Overall, this seems like a good system. It is a very complex situation for an API to support, but I think you provided enough generality with the structure that a pretty wide array of formats and use cases could be supported!